geomesa-blobstore/geomesa-blobstore-handlers/geomesa-blobstore-gdal-handler/src/main/scala/org/locationtech/geomesa/blob/handlers/gdal/GDALFileHandler.scala

```diff
@@ -0,0 +1,107 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.blob.handlers.gdal
+
+import java.io.File
+import java.lang.Boolean
+import java.util
+
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.{Coordinate, Geometry, GeometryFactory}
+import org.gdal.gdal.{Dataset, Transformer, gdal}
+import org.gdal.gdalconst.gdalconstConstants
+import org.locationtech.geomesa.blob.core.handlers.AbstractFileHandler
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+/**
+  * GdalFileHandler uses GDAL to load metadata from candidate files to determine if
+  * they can be ingested into the blob store.
+  *
+  */
+class GDALFileHandler extends AbstractFileHandler with LazyLogging {
+  import GDALFileHandler._
+
+  val geomFactory = new GeometryFactory()
+  val initialized = Try(gdal.AllRegister()) match {
+    case Success(v) =>
+      logger.info("Successfully loaded GDAL native interface bindings")
+      true
+    case Failure(e) =>
+      logger.error("Failed to load GDAL native libraries in GeoMesa GDAL FileHandler, Ensure that the path of the native GDAL libraries and JNI libraries are in the LD_LIBRARY_PATH environment variable", e)
+      false
+  }
+
+  override def getGeometryFromFile(file: File): Option[Geometry] = {
+    if (!initialized) {
+      None
+    } else {
+      val nitfFile = gdal.Open(file.getAbsolutePath, gdalconstConstants.GA_ReadOnly)
+      val geom = getImageBoundsGeometry(nitfFile)
+      nitfFile.delete()
```

Comment:
it sounds just like a close(). Typical pattern is:
open resource
try { stuff } finall { close resource }
please update